pkg/ruler/base/api_test.go

```diff
@@ -163,6 +164,53 @@ func TestRuler_alerts(t *testing.T) {
 	require.Equal(t, string(expectedResponse), string(body))
 }
 
+func TestRuler_GetRulesLabelFilter(t *testing.T) {
+	cfg := defaultRulerConfig(t, newMockRuleStore(mockRules))
+
+	r := newTestRuler(t, cfg)
+	defer r.StopAsync()
+
+	a := NewAPI(r, r.store, log.NewNopLogger())
+
+	tc := []struct {
+		name     string
+		URLQuery string
+		output   string
+		err      error
+	}{
+		{
+			name:   "query with no filters",
+			output: "test:\n    - name: group1\n      interval: 1m\n      rules:\n        - record: UP_RULE\n          expr: up\n        - alert: UP_ALERT\n          expr: up < 1\n          labels:\n            foo: bar\n        - alert: DOWN_ALERT\n          expr: down < 1\n          labels:\n            namespace: delta\n",
+		},
+		{
+			name:     "query with a valid filter found",
+			URLQuery: "labels=namespace:delta,foo:bar",
+			output:   "test:\n    - name: group1\n      interval: 1m\n      rules:\n        - alert: UP_ALERT\n          expr: up < 1\n          labels:\n            foo: bar\n        - alert: DOWN_ALERT\n          expr: down < 1\n          labels:\n            namespace: delta\n",
+		},
+		{
+			name:     "query with an invalid query param",
```

Comment:
I'm not sure how to read this test-case. Should it just return the default list if the param is empty?

Reply:
Here we have a query param that is not recognized/handled by the API, so it's just ignored, that's why it's returning the result as if no query param was specified.

Reply:
I think if you defined a const with the default rule group which returns both for `query with no filters` and this test case, it'll read a lot more clearly.